pkg/controller/storagecluster/external_resources.go

```diff
@@ -37,6 +38,36 @@ type ExternalResource struct {
 	Name string            `json:"name"`
 }
 
+// filterUnavailableStorageClasses removes SC's that are not present in the ExternalResource array
+func (r *ReconcileStorageCluster) filterUnavailableStorageClasses(scs []*storagev1.StorageClass, data []ExternalResource) (filteredScs []*storagev1.StorageClass, disableCephFS bool) {
+	disableCephFS = true
+	// RBD should always be present
+	filteredScs = append(filteredScs, scs[1])
+	for _, d := range data {
```

Comment:
Don't cycle through all the data. Cycle through `scs`, check which StorageClassName const they match, and append to filteredScs that way.